### PR TITLE
test: allow individual testing of e2e test cases through env variable

### PR DESCRIPTION
**Pull request by sunpa93**
**What type of PR is this?**

**What this PR does / why we need it**:
- allows testing for individual test cases for those with specific focus by setting env variable for GINKGO_FOCUS
- if GINKGO_FOCUS env variable is set, it will override topology

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -48,11 +48,16 @@ SCHEDULER_EXTENDER_LDFLAGS ?= "-X ${PKG}/pkg/azuredisk.schedulerVersion=${IMAGE_
 LDFLAGS ?= "-X ${PKG}/pkg/azuredisk.driverVersion=${IMAGE_VERSION} -X ${PKG}/pkg/azuredisk.gitCommit=${GIT_COMMIT} -X ${PKG}/pkg/azuredisk.buildDate=${BUILD_DATE} -X ${PKG}/pkg/azuredisk.DriverName=${DRIVER_NAME} -X ${PKG}/pkg/azuredisk.topologyKey=${TOPOLOGY_KEY} -extldflags "-static"" ${GOTAGS}
 E2E_HELM_OPTIONS ?= --set image.azuredisk.repository=$(REGISTRY)/$(IMAGE_NAME) --set image.azuredisk.tag=$(IMAGE_VERSION) --set image.azuredisk.pullPolicy=Always --set image.schedulerExtender.repository=$(REGISTRY)/$(SCHEDULER_EXTENDER_IMAGE_NAME) --set image.schedulerExtender.tag=$(IMAGE_VERSION) --set image.schedulerExtender.pullPolicy=Always ${ADDITIONAL_E2E_HELM_OPTIONS}
 GINKGO_FLAGS = -ginkgo.v
+# If one needs to set focus to specific test cases, set GINKGO_FOCUS env var
+ifneq ($(GINKGO_FOCUS),)
+GINKGO_FLAGS += -ginkgo.focus=${GINKGO_FOCUS}
+else
 ifeq ($(ENABLE_TOPOLOGY), true)
```
Comment (landreasyan) [inline Makefile:55]: np: same line - `else ifeq`

**Review by landreasyan — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -48,11 +48,16 @@ SCHEDULER_EXTENDER_LDFLAGS ?= "-X ${PKG}/pkg/azuredisk.schedulerVersion=${IMAGE_
...
 GINKGO_FLAGS += -ginkgo.focus="\[multi-az\]"
 else
 GINKGO_FLAGS += -ginkgo.focus="\[single-az\]"
 endif
+endif
```
Comment (landreasyan) [inline Makefile:60]: no need for the additional `endif`

Comment (sunpa93) [reply Makefile:60]: `endif` on line 59 is for closing if statement on 55 and one on line 60 is for 52.

Comment (sunpa93) [reply Makefile:60]: ah I saw this comment alone without the context above and was wondering what you meant

**Comment by edreed** [comment]
Why is this being added when you can run the e2e test with a custom focus via the command line?

**Pull request closed without merging**